### PR TITLE
feat: private claims (incl. serialisation refactoring and jwt profile fix)

### PR DESCRIPTION
**Pull request by livio-a**

### CODE REVIEW

File: pkg/oidc/token.go [hunk 1]
```diff
@@ -3,331 +3,433 @@ package oidc
...
-		Region:        j.JsonUserinfoAddress.Region,
-		StreetAddress: j.JsonUserinfoAddress.StreetAddress,
+	return NewJWTProfileAssertion(keyData.UserID, keyData.KeyID, audience, []byte(keyData.Key)), nil
+}
+
+func NewJWTProfileAssertion(userID, keyID string, audience []string, key []byte) *JWTProfileAssertion {
+	return &JWTProfileAssertion{
+		PrivateKey:   key,
+		PrivateKeyID: keyID,
+		Issuer:       userID,
```
Comment (fgerschwiler) [inline pkg/oidc/token.go:428]: userID in issuer?

File: pkg/oidc/token_request.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+type JWTTokenRequest struct {
+	Issuer    string   `json:"iss"`
+	Subject   string   `json:"sub"`
+	Scopes    Scopes   `json:"-"`
+	Audience  Audience `json:"aud"`
+	IssuedAt  Time     `json:"iat"`
+	ExpiresAt Time     `json:"exp"`
+}
+
+//GetSubject implements the Claims interface
```
Comment (fgerschwiler) [inline pkg/oidc/token_request.go:46]:
```suggestion
//GetIssuer implements the Claims interface
```

File: pkg/op/storage.go [hunk 3]
```diff
@@ -28,8 +28,9 @@ type AuthStorage interface {
 type OPStorage interface {
 	GetClientByClientID(context.Context, string) (Client, error)
 	AuthorizeClientIDSecret(context.Context, string, string) error
-	GetUserinfoFromScopes(context.Context, string, []string) (*oidc.Userinfo, error)
-	GetUserinfoFromToken(ctx context.Context, tokenID, subject, origin string) (*oidc.Userinfo, error)
+	GetUserinfoFromScopes(context.Context, string, string, []string) (oidc.UserInfo, error)
```
Comment (fgerschwiler) [inline pkg/op/storage.go:31]: what are this parameters?

File: pkg/utils/marshal.go [hunk 4]
```diff
@@ -19,3 +21,15 @@ func MarshalJSON(w http.ResponseWriter, i interface{}) {
 		logrus.Error("error writing response")
 	}
 }
+
+func ConcatenateJSON(first, second []byte) ([]byte, error) {
```
Comment (fgerschwiler) [inline pkg/utils/marshal.go:25]: it would be nice to have a test for this

**Review by fgerschwiler — CHANGES_REQUESTED** [review]

File: pkg/oidc/token.go (hunk 1 above)
Comment (livio-a) [reply pkg/oidc/token.go:428]: jwt profile is done by service accounts... so they're actually the issuer

**Review by fgerschwiler — APPROVED** [review]

**Pull request merged**